### PR TITLE
Added `disable_service_types` option to protobuf-ts/plugin for disabling service metadata generation

### PR DESCRIPTION
**Pull request by ColinLaws**
## Summary of changes
Small change to introduce optionality in generating service metadata for proto messages. I have named the option `disable_service_types`. 

## Precaution
I am having difficulty understanding how to run tests after following the guide provided in `CONTRIBUTING.md`. I was not able to run tests successfully against `packages/plugin`, as I have an error that `@protobuf-ts/plugin-framework` is missing. I made sure to build and test the plugin-framework, but still I encounter issues when trying to run tests for the plugin package. Any help in contributing is appreciated, I'd like to make sure I'm not introducing bugs into this library. Thanks everyone.

### CODE REVIEW

**Comment by ColinLaws** [comment]
Hey @timostamm I was wondering if the option name `disable_service_types` was okay with you. The reason I pluralized "types" and added some extra spacing between each word was for clarity sake when reading. I did the same for any variables/keys of objects, I tried to keep the code as cohesive as possible with the conventions that have already been established within the project.

File: packages/plugin/src/protobufts-plugin.ts [hunk 1]
```diff
@@ -94,6 +94,9 @@ export class ProtobuftsPlugin extends PluginBase {
                 "the default behaviour, this option has no effect.",
             excludes: ['eslint_disable'],
         },
+        disable_service_types: {
+            description: 'Prevents the generation of service metadata.',
+        },
```
Comment (timostamm) [inline packages/plugin/src/protobufts-plugin.ts:99]: There is a caveat to this option: The service metadata is used by generated clients, and the only way to disable clients completely is to set the option `force_client_none`. `client_none` is not enough. The user could have `(ts.client) = GRPC1_CLIENT` in a proto file.

Generating code that cannot compile could be considered a bug, so I wonder how we could avoid that.

Maybe the option should be named `force_disable_services`, and do the same as `force_client_none` and `force_server_none`. The option definition could use an `excludes` array to automatically disallow options like `client_grpc1` that do not make sense to be set at the same time.

**Comment by timostamm** [comment]
> I am having difficulty understanding how to run tests after following the guide provided in `CONTRIBUTING.md`. I was not able to run tests successfully against `packages/plugin`, as I have an error that `@protobuf-ts/plugin-framework` is missing. I made sure to build and test the plugin-framework, but still I encounter issues when trying to run tests for the plugin package.

This is likely caused by lerna, the tool to link the packages within this monorepo. Please try to run `make lerna-bootstrap` in the root directory of the project. In general, just running `make` should runs all builds and all test, but you will need to have all build tools mentioned installed.

File: packages/plugin/src/protobufts-plugin.ts (hunk 1 above)
Comment (timostamm) [reply packages/plugin/src/protobufts-plugin.ts:99]: @ColinLaws, what do you think about this caveat, and the proposed fix?

Comment (ColinLaws) [reply packages/plugin/src/protobufts-plugin.ts:99]: Sorry for leaving this pull request Hanging @timostamm My day job working on a project that is dependent on your package is taking up nearly all of my time. 

I see the issue you're mentioning, and I do agree with your suggested solution to resolving this. If someone is consuming this package and does not wish to generate the services, then I agree it's a good bet that they aren't going to write code that is dependent on them. I can try implementing a fix for this likely sometime this week. I may be able to fit it into what I'm doing at work, as this is the only package keeping us from upgrading to NG13.

Comment (timostamm) [reply packages/plugin/src/protobufts-plugin.ts:102]: Thank you @ColinLaws, looking forward to it :)

**Comment by tannera-cargill** [comment]
This is also a feature that we need, since we only are using protobuf-ts to generate the TS types and encoding/decoding functions.

File: packages/plugin/src/protobufts-plugin.ts [hunk 2]
```diff
@@ -97,6 +97,10 @@ export class ProtobuftsPlugin extends PluginBase {
                 "the default behaviour, this option has no effect.",
             excludes: ['eslint_disable'],
         },
+        force_disable_services: {
+            description: 'Do not generate GRPC services, clients, or servers.',
```
Comment (timostamm) [inline packages/plugin/src/protobufts-plugin.ts:101]: We should give a bit more details about what this option does, for example:

```suggestion
            description: "Do not generate anything for service definitions, and \n" +
                         "ignore options in proto files. This is the same as setting both \n" + 
                         "`force_server_none` and `force_client_none`, but also stops \n" + 
                         "generating service metadata."
```

File: packages/plugin/src/our-options.ts [hunk 3]
```diff
@@ -328,6 +331,9 @@ export function makeInternalOptions(
     if (params?.add_pb_suffix) {
         o.addPbSuffix = true;
     }
+    if (params?.force_disable_services) {
+      o.forceDisableServices = true;
+    }
```
Comment (timostamm) [inline packages/plugin/src/our-options.ts:336]: I wonder if we should set `forcedServerStyle` and `forcedClientStyle` here to be explicit, but can't think of a case the very obvious if-statement below wouldn't catch.

**Review by timostamm — APPROVED** [review]
@ColinLaws, thanks for the update! I think we need to explain the option with a bit more detail, but that should be it.

**Comment by timostamm** [comment]
Released in [v2.7.0](https://github.com/timostamm/protobuf-ts/releases/tag/v2.7.0).

**Pull request merged**